#288: Added sh:SelectExpression, with test cases and TTL changes

HolgerKnublauch opened the pull request.

HolgerKnublauch committed:
#288: Added sh:SelectExpression, with test cases and TTL changes

ajnelson-nist committed:
#222, #288: Update SHACL-SHACL for general NodeExpression and SelectExpression

Signed-off-by: Alex Nelson <alexander.nelson@nist.gov>

ajnelson-nist commented:
@HolgerKnublauch - I added an update to SHACL-SHACL.  It seems one update should have gone in for #274 : If `sh:targetNode` can now allow blank nodes in their object position, then it seems there's currently no `sh:PropertyShape` necessary because there's no constraint to specify.  Does this need to be broken out into a separate PR, possibly also to address any updates to `targetNode-nodeKind` text?

bergos reviewed (approved):
LGTM, as mentioned in #301, we may need to revisit the `sh:path` property. But that should be done in a separate PR.

ajnelson-nist commented:
Oh dear.  For awareness, there is a comment thread tied to just the last commit, [here](https://github.com/w3c/data-shapes/commit/8632a3e0067c5bfde8f70de79a669b1e3d95e9f5), documenting that I'm going to undo the SHACL-SHACL changes and move them to another PR.  It's not displaying in this PR page.  Sorry @bergos , you'll need to re-review.

ajnelson-nist committed:
#222, #288: Undo update SHACL-SHACL for general NodeExpression and SelectExpression

This patch undoes commit 8632a3e.  This was done manually instead of with `git revert`.

Signed-off-by: Alex Nelson <alexander.nelson@nist.gov>

ajnelson-nist commented:
Ok, the updates to SHACL-SHACL I had here are now moved out-of-band to #302 .

TallTed commented on shacl12-vocabularies/shacl.ttl:
```diff
@@ -1174,6 +1176,23 @@ sh:xone
+sh:keyParameter
+    a rdf:Property ;
+	rdfs:label "key parameter"@en ;
+	rdfs:comment "Set to true for the parameter of a node expression function that unique identifies it."@en ;
```
```suggestion
	rdfs:comment "Set to true for the parameter of a node expression function that uniquely identifies it."@en ;
```

TallTed reviewed (dismissed):
typo

HolgerKnublauch committed:
Update shacl12-vocabularies/shacl.ttl

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

HolgerKnublauch commented:
@TallTed could you kindly recheck whether you can unblock this PR? Thanks.

afs reviewed (approved):
This looks good for a FPWD.
I can't check the test case (no code).
There details recorded elsewhere for the example using sh:path.

HolgerKnublauch commented:
@TallTed this PR is still blocked because you have requested changes. As I have long applied your suggestion, I believe you still need to change the status of your review away from Request Changes.

(I don't think simple change requests such as typo fixes should ever block progress)

caribouW3 commented:
> @TallTed this PR is still blocked because you have requested changes. As I have long applied your suggestion, I believe you still need to change the status of your review away from Request Changes.
> 
> (I don't think simple change requests such as typo fixes should ever block progress)

Since that request has been already accepted, I'll dismiss the review comment to unlock merging.

HolgerKnublauch commented:
Thanks a lot, @caribouW3. I am going to merge this PR into the main branch now because it has sufficient approvals and has been widely agreed to be done by the group when #222 was accepted. The Core spec even contains a reference to this already, so it's best to get this consistent now. Also, there are follow-up branches based on this already piling up.
